flit/validate.py

```diff
@@ -0,0 +1,335 @@
+"""Validate various pieces of packaging data"""
+
+import logging
+import os
+from pathlib import Path
+import re
+import requests
+import sys
+
+from .common import InvalidVersion
+
+log = logging.getLogger(__name__)
+
+def get_cache_dir() -> Path:
+    """Locate a platform-appropriate cache directory for flit to use
+
+    Does not ensure that the cache directory exists.
+    """
+    if os.name == 'posix' and sys.platform != 'darwin':
+        # Linux, Unix, AIX, etc.
+        # use ~/.cache if empty OR not set
+        xdg = os.environ.get("XDG_CACHE_HOME", None) or (
+        os.path.expanduser('~/.cache'))
+        return Path(xdg, 'flit')
+
+    elif sys.platform == 'darwin':
+        return Path(os.path.expanduser('~'), 'Library/Caches/flit')
+
+    else:
+        # Windows (hopefully)
+        local = os.environ.get('LOCALAPPDATA', None) or (
+        os.path.expanduser('~\\AppData\\Local'))
+        return Path(local, 'flit')
+
+def _verify_classifiers_cached(classifiers):
+    """Check classifiers against the downloaded list of known classifiers"""
+    with (get_cache_dir() / 'classifiers.lst').open(encoding='utf-8') as f:
+        valid_classifiers = set(l.strip() for l in f)
+
+    invalid = classifiers - valid_classifiers
+    return ["Unrecognised classifier: {!r}".format(c)
+            for c in sorted(invalid)]
+
+
+def _download_classifiers():
+    """Get the list of valid trove classifiers from PyPI"""
+    log.info('Fetching list of valid trove classifiers')
+    resp = requests.get(
+        'https://pypi.python.org/pypi?%3Aaction=list_classifiers')
+    resp.raise_for_status()
```

Comment:
possibly find the corresponding URL for warehouse, and use as a fallback ?

Reply:
I've found https://pypi.org/pypi?%3Aaction=list_classifiers , but I'm asking whether it's something we can rely on before I switch to it: https://github.com/pypa/warehouse/issues/1300

Reply:
Donald says it's safe to rely on, so I've switched to that.

Reply:
Good !